### PR TITLE
Add beautifulsoup4

### PR DESCRIPTION
**Pull request by jschueller**
Fixes https://github.com/conda-forge/staged-recipes/issues/2227

### CODE REVIEW

File: recipes/beautifulsoup4/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - setuptools
+  run:
+    - python
+
+test:
+  imports:
+    - bs4
+
+about:
+  home: http://www.crummy.com/software/BeautifulSoup/index.html
```
Comment (jakirkham) [inline recipes/beautifulsoup4/meta.yaml]: They seem to be using [1] on [PyPI]( https://pypi.python.org/pypi/beautifulsoup4/4.5.3 ). Though it redirects to [2], which seems the same as [3] proposed here.

1 - http://www.crummy.com/software/BeautifulSoup/bs4/
2 - https://www.crummy.com/software/BeautifulSoup/
3 - http://www.crummy.com/software/BeautifulSoup/index.html

File: recipes/beautifulsoup4/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+{% set name = "beautifulsoup4" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  md5: 937e0df0d699a1237646f38fd567f0c6
```
Comment (jakirkham) [inline recipes/beautifulsoup4/meta.yaml]: Can get `sha256` from [PyPI]( https://pypi.org/project/beautifulsoup4/#files ).

File: recipes/beautifulsoup4/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+build:
+  number: 0
+  script:
+    - python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (jakirkham) [inline recipes/beautifulsoup4/meta.yaml]: Can put on one line with `script` and no list for simplicity.

**Comment by jakirkham** [comment]
Looks pretty good. Some minor comments above.

**Comment by jschueller** [comment]
@jakirkham ok

**Pull request merged**